crates/bevy_asset/src/saver.rs

```diff
@@ -8,6 +8,11 @@ use std::{borrow::Borrow, hash::Hash, ops::Deref};
 
 /// Saves an [`Asset`] of a given [`AssetSaver::Asset`] type. [`AssetSaver::OutputLoader`] will then be used to load the saved asset
 /// in the final deployed application. The saver should produce asset bytes in a format that [`AssetSaver::OutputLoader`] can read.
+///
+/// This trait defines asset-specific logic to convert Rust types into serializable data.
```

Comment:
Not quite, it's more like:
* AssetReader: Read bytes from <somewhere> 
* AssetWriter: Write bytes to <somewhere>
* AssetLoader: Read bytes from an AssetReader and turn into a T: Asset
* AssetSaver: Turn T: Asset into bytes and write them to an AssetWriter

Writing this out, I feel like AssetReader/AssetWriter should be renamed to something more general. Unless I got these types wrong, and the byte stuff is another trait? Idk actually.